### PR TITLE
po/remove deferred preset

### PR DESCRIPTION
**Pull request by TurboGit**
A second version based on @dterrahe PR and extending to fix the copy/paste when using presets.

At the same time some issues have been fixed.

Not fully ready as I still need to check the style apply.

### CODE REVIEW

**Comment by TurboGit** [comment]
I think this last version also fix the support for styles. Another good journey into the presets/styles copy/duplicate of modules. All in all some issues have been fixed while working this module name based on presets.

File: src/develop/develop.c [hunk 1]
```diff
@@ -1122,8 +1158,12 @@ static void _dev_add_history_item(dt_develop_t *dev,
 {
   if(!darktable.gui || darktable.gui->reset) return;
 
+  gchar *saved_name = g_strdup(module->multi_name);
```
Comment (dterrahe) [inline src/develop/develop.c:1161]: @TurboGit this needs to be saved _before_ it gets potentially changed by `_dev_auto_module_label`. Now the strcmp below always returns 0, so `target` is always passed to `_dev_undo_start_record_target` causing updates of the history list  in the left panel to be skipped on multiple changes of the same widget and the multi-name shown there to get out of sync with the name on the module itself.

Comment (TurboGit) [reply src/develop/develop.c:1161]: Arrrrgggg :( Code refactoring with side-effects is not easy and I completely missed this. Will fix, and of course thanks for catching.

Comment (TurboGit) [reply src/develop/develop.c:1161]: Now fixed, do not hesitate to double check :)

**Pull request merged**